modules/wallet/encrypt.go

```diff
@@ -52,30 +52,15 @@ func checkMasterKey(tx *bolt.Tx, masterKey crypto.TwofishKey) error {
 	return verifyEncryption(uk, encryptedVerification)
 }
 
-// initEncryption checks that the provided encryption key is the valid
-// encryption key for the wallet. If encryption has not yet been established
-// for the wallet, an encryption key is created.
-func (w *Wallet) initEncryption(masterKey crypto.TwofishKey) (modules.Seed, error) {
-	// Create a random seed.
-	var seed modules.Seed
-	_, err := rand.Read(seed[:])
-	if err != nil {
-		return modules.Seed{}, err
-	}
-
-	err = w.db.Update(func(tx *bolt.Tx) error {
+// initEncryption initializes and encrypts the primary SeedFile.
+func (w *Wallet) initEncryption(masterKey crypto.TwofishKey, seed modules.Seed) (modules.Seed, error) {
+	err := w.db.Update(func(tx *bolt.Tx) error {
 		wb := tx.Bucket(bucketWallet)
 		// Check if the wallet encryption key has already been set.
 		if wb.Get(keyEncryptionVerification) != nil {
 			return errReencrypt
 		}
 
-		// If the input key is blank, use the hash of the seed to create the
-		// master key. Otherwise, use the input key.
-		if masterKey == (crypto.TwofishKey{}) {
-			masterKey = crypto.TwofishKey(crypto.HashObject(seed))
-		}
```

Comment:
oh I see you moved the code.

To me it makes it feel like it's more likely that initEncryption could be used incorrectly. hmm